### PR TITLE
feat: npm to pnpm

### LINKED ISSUE #34: Support pnpm
Hello,

It would be great if this library could support `pnpm`! In other words, the signature for the export should be changed from:

```ts
export default function convert(str: string, to: 'npm' | 'yarn'): string;
```

To:

```ts
export default function convert(str: string, to: 'npm' | 'yarn' | 'pnpm'): string;
```

From what I can gather, the syntax of pnpm is mostly the same as that of yarn. So this should be pretty easy.

For example:

- `npm install foo --save` --> `yarn add foo`
- `npm install foo --save` --> `pnpm add foo`

It's the same.

And:

- `npm install foo -D` --> `yarn add foo -D`
- `npm install foo -D` --> `pnpm add foo -D`

It's also the same.

However, there are some differences. For example:

- `npm install foo --save-dev` --> `yarn add foo --dev`
- `npm install foo --save-dev` --> `pnpm add foo --save-dev`

I'm by no means an expert in pnpm, so I'm sure there are more differences.

### PR DESCRIPTION
**Pull request by armano2**
Add support for all command that are support in npm to yarn to work with pnpm

Changes:
- add support for `npm` to `pnpm`
- add missing conversion for `npm outdated`
- add missing conversion for `npm pack`
- add more tests for `npm` to `yarn`

fixes #34 ref https://github.com/babel/website/pull/2725

### CODE REVIEW

**Comment by bgub** [comment]
Thank you for another helpful PR! I just merged #36, so you can rebase it now.

File: test/index.spec.ts [hunk 1]
```diff
@@ -2,183 +2,180 @@
...
+  describe('to Yarn', () => {
+    it.each(tests)('%s', (npmValue, yarnValue) => {
+      expect(convert(npmValue, 'yarn')).toEqual(yarnValue)
+    })
+  })
+
+  describe('to PNPM', () => {
+    it.each(tests)('%s', (npmValue, _yarnValue, pnpmValue) => {
+      expect(convert(npmValue, 'pnpm')).toEqual(pnpmValue)
+    })
```
Comment (armano2) [inline test/index.spec.ts:200]: I converted all tests to be executed in loop, to make sure that if we add new tests, we validate it against yarn and pnpm

**Comment by armano2** [comment]
@nebrelbug rebased, i noticed few additional issues with converting and fixed them

I also added few new test cases

**Comment by liuxingbaoyu** [comment]
Wow! Amazing work, thank you so much!

**Comment by nicolo-ribaudo** [comment]
Thank you! :)

**Comment by bgub** [comment]
@armano2 thank you! I'll merge this now.

**Comment by bgub** [comment]
I just released this as part of version 2.0.0!

**Comment by Zamiell** [comment]
thanks armano!

**Pull request merged**